numpy/linalg/linalg.py

```diff
@@ -1568,10 +1572,13 @@ def matrix_rank(M, tol=None):
     M = asarray(M)
     if M.ndim < 2:
         return int(not all(M==0))
-    S = svd(M, compute_uv=False)
+    if hermitian:
+        S = eigvalsh(M)
+    else:
+        S = abs(svd(M, compute_uv=False))
```

Comment:
See [this comment](https://github.com/numpy/numpy/pull/8557#issuecomment-277075209) for the derivation.